linux/src/async_socket_linux.c

```diff
@@ -257,6 +261,10 @@ static void event_complete_callback(void* context, COMPLETION_PORT_EPOLL_ACTION
                 break;
             }
         }
+        if (interlocked_decrement(event_complete_executing) == 0)
+        {
+            wake_by_address_single(event_complete_executing);
```

Comment:
>wake_by_address_single [](http://example.com/codeflow?start=12&length=22)

The only time we're listening for this number of events to reach "0" is in `ASYNC_SOCKET_LINUX_STATE_CLOSING` state.

I think it is better to only wake up in that state.